app/controllers/api/events_controller.rb

```diff
@@ -0,0 +1,36 @@
+class API::EventsController < ApplicationController
+   skip_before_action :verify_authenticity_token
+
+   before_action :set_access_control_headers
+
+   def set_access_control_headers
+     headers['Access-Control-Allow-Origin'] = '*'
+     headers['Access-Control-Allow-Methods'] = 'POST, GET, OPTIONS'
+     headers['Access-Control-Allow-Headers'] = 'Content-Type'
+   end
+ 
+   def create
+   	registered_application = RegisteredApplication.find_by(url: request.env['HTTP_ORIGIN'])
+
+   	if registered_application.nil?
```

Comment:
You could make this a guard clause by putting it into one line. Like return render json: { error: "Unregistered application"}, status: :unprocessable_entity if registered_application.nil?